packages/tailwindcss-language-server/src/project-locator.ts

```diff
@@ -510,25 +519,37 @@ async function* contentSelectorsFromCssConfig(entry: ConfigEntry): AsyncIterable
   }
 }
 
-async function* detectContentFiles(base: string): AsyncIterable<string> {
+async function* detectContentFiles(
+  base: string,
+  inputFile,
+  sources: string[],
+): AsyncIterable<string> {
   try {
     let oxidePath = resolveFrom(path.dirname(base), '@tailwindcss/oxide')
     oxidePath = pathToFileURL(oxidePath).href
 
-    const oxide: typeof import('@tailwindcss/oxide') = await import(oxidePath)
+    const oxide: typeof import('@tailwindcss/oxide') = await import(oxidePath).then(
+      (o) => o.default || o,
+    )
 
     // This isn't a v4 project
     if (!oxide.scanDir) return
 
-    let { files, globs } = oxide.scanDir({ base, globs: true })
+    let { files, globs, candidates } = oxide.scanDir({
+      base,
+      sources: sources.map((pattern) => ({
+        base: path.dirname(inputFile),
+        pattern,
+      })),
+    })
 
     for (let file of files) {
       yield normalizePath(file)
     }
 
-    for (let { base, glob } of globs) {
+    for (let { base, pattern } of globs) {
       // Do not normalize the glob itself as it may contain escape sequences
-      yield normalizePath(base) + '/' + glob
+      yield normalizePath(base) + '/' + pattern
```

Comment:
This is going to break older versions of oxide with the extension. Should we patch over it until we release a more stable version?

Reply:
Yes, we should. The first version should basically call the "modern" scanDir and then if it throws call `oxide.scanDir({ base, globs: true })` and map the globs so it's compatible.